src/component.rs

```diff
@@ -81,25 +79,25 @@ impl Deref for TrailerSeparator {
         match self {
             TrailerSeparator::ColonSpace => ": ",
             TrailerSeparator::SpacePound => " #",
+            TrailerSeparator::__NonExhaustive => "",
         }
     }
 }
 
 impl fmt::Display for TrailerSeparator {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            TrailerSeparator::ColonSpace => f.write_str(": "),
-            TrailerSeparator::SpacePound => f.write_str(" #"),
-        }
+        f.write_str(self)
```

Comment:
Ah! Of course, much better. Thank you for this, makes sense, and good to have in my tool belt from now on.

Reply:
However, this makes me go "hmm" in terms of this being a strictly typed API.  We should probably only implement AsRef and not Deref to avoid implicit conversions like this.

Reply:
Wait, nm, I'm thinking for everything but `TrailerSeparator`.  I think it is fine for `TrailerSeparator` to do this.